src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -58,16 +102,26 @@ const HASH_SALT_ROUNDS = 1
  */
 export const hashFieldValues = (
   prefilledFormFields: PossiblyPrefilledField[],
+  childrenBirthRecords?: MyInfoChildData,
 ): MyInfoHashPromises => {
   const readOnlyHashPromises: MyInfoHashPromises = {}
 
   prefilledFormFields.forEach((field) => {
+    // For children fields, we need to explode their subfields.
+    if (
+      isMyInfoChildrenBirthRecords(field.myInfo?.attr) &&
+      childrenBirthRecords !== undefined
+    ) {
+      hashChildrenFieldValues(field, childrenBirthRecords, readOnlyHashPromises)
```

Comment:
/nit I'm not a fan of passing in a object to be mutated, I would have probably kept `hashChildrenFieldValues()` to be a pure function and did the augmentation of `readOnlyHashPromises` here:
```suggestion
      Object.assign(
        readOnlyHashPromises,
        hashChildrenFieldValues(field, childrenBirthRecords)
      )
```

[Edit] I retract the comment. After seeing `compareHashedValues()` below, which is similar but harder to change, I think it's better to have the 2 similar functions be consistent in their processing.